### PR TITLE
[KDESKTOP-1252] Handle the Windows ERROR_BAD_NET_PATH just like FILE_NOT_FOUND

### PR DESCRIPTION
**Pull request by luc-guyot-infomaniak**

### CODE REVIEW

File: src/common/utility_win.cpp [hunk 1]
```diff
@@ -36,43 +34,9 @@
 
 static const char systemRunPathC[] = "HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows\\CurrentVersion\\Run";
 static const char runPathC[] = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Run";
-static const char themePathC[] = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
-static const char lightThemeKeyC[] = "SystemUsesLightTheme";
 
 namespace KDC {
 
-static void setupFavLink_private(const QString &folder) {
```
Comment (luc-guyot-infomaniak) [inline src/common/utility_win.cpp:44]: Unused code detected by QtCreator.

File: src/libcommon/utility/utility.h [hunk 2]
```diff
@@ -117,6 +117,14 @@ struct COMMON_EXPORT CommonUtility {
         static void writeSignalFile(AppType appType, SignalType signalType) noexcept;
         static void clearSignalFile(AppType appType, SignalCategory signalCategory, SignalType &signalType) noexcept;
 
+        static bool fileExists(const std::error_code &ec) noexcept;
+
+#ifdef _WIN32
+        static std::wstring getErrorMessage(DWORD dwordError);
+        static std::wstring getLastErrorMessage();
+        static bool fileExists(DWORD dwError) noexcept;
```
Comment (ClementKunz) [inline src/libcommon/utility/utility.h:125]: Same here, can we find a better name?

File: extensions/windows/cfapi/Common/utilities.cpp [hunk 3]
```diff
@@ -324,18 +324,29 @@ bool Utilities::readNextValue(std::wstring &message, std::wstring &value) {
     return true;
 }
 
+namespace {
+
+bool fileExists(DWORD dwError) noexcept {
```
[the remark this answers is not in the document]
Comment (luc-guyot-infomaniak) [reply extensions/windows/cfapi/Common/utilities.cpp:329]: I have opted for `isLikeFileNotFoundError`.    Done.

File: src/libcommon/utility/utility.h (hunk 2 above)
Comment (luc-guyot-infomaniak) [reply src/libcommon/utility/utility.h]: Same modification as above. Done.

**Review by ClementKunz — DISMISSED** [review]

File: extensions/windows/cfapi/Common/utilities.cpp [hunk 4]
```diff
@@ -17,12 +17,12 @@
  */
 
 #include "utilities.h"
+#include "../../../../src/libcommon/utility/utility_base.h"
```
Comment (ClementKunz) [inline extensions/windows/cfapi/Common/utilities.cpp:20]: Is the part `../../../../src/`really necessary?

Comment (ChristopheLarchier) [reply extensions/windows/cfapi/Common/utilities.cpp:20]: Try to link the extension with libcommon library.

Comment (luc-guyot-infomaniak) [reply extensions/windows/cfapi/Common/utilities.cpp:20]: Would it be OK to keep things like this and create a common library that can be shared with extensions in a second step? I have written the following Jira ticket : https://infomaniak.atlassian.net/browse/KDESKTOP-1435 regarding the creation of a new utility library.

Comment (ChristopheLarchier) [reply extensions/windows/cfapi/Common/utilities.cpp:20]: OK

**Review by ChristopheLarchier — APPROVED** [review]

**Pull request merged**